query/src/main/scala/hmda/query/projections/institutions/InstitutionView.scala

```diff
@@ -0,0 +1,101 @@
+package hmda.query.projections.institutions
+
+import akka.actor.{ ActorRef, ActorSystem, Props }
+import akka.persistence.{ RecoveryCompleted, SnapshotOffer }
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Sink
+import hmda.model.institution.Agency.CFPB
+import hmda.model.institution.InstitutionType.Bank
+import hmda.model.institution.{ Inactive, Institution }
+import hmda.persistence.messages.CommonMessages.{ Command, Event, GetState, Shutdown }
+import hmda.persistence.messages.events.institutions.InstitutionEvents._
+import hmda.persistence.model.HmdaPersistentActor
+import hmda.persistence.processing.HmdaQuery._
+
+object InstitutionView {
+
+  val name = "institutions-view"
+
+  case class GetInstitutionById(institutionId: String) extends Command
+  case class GetInstitutionsById(ids: List[String]) extends Command
+  case class LastProcessedEventOffset(seqNr: Long)
+  case object StreamCompleted
+
+  def props(): Props = Props(new InstitutionView)
+
+  def createInstitutionQuery(system: ActorSystem): ActorRef = {
+    system.actorOf(InstitutionView.props(), "institutions-view")
+  }
+
+  case class InstitutionViewState(institutions: Set[Institution] = Set.empty[Institution], seqNr: Long = 0L) {
+    def updated(event: Event): InstitutionViewState = {
+      event match {
+        case InstitutionCreated(i) =>
+          InstitutionViewState(institutions + i, seqNr)
+        case InstitutionModified(i) =>
+          val others = institutions.filterNot(_.id == i.id)
+          InstitutionViewState(others + i, seqNr)
+      }
+    }
+  }
+
+}
+
+class InstitutionView extends HmdaPersistentActor {
```

Comment:
Oh, and in case you are missing some design there this is a known issue, the institutions model is not complete. For example the SQL view I'm referring to can't be done with the data that we have right now, but it will be there in the future (i.e. we will know when a create / modify event happened for an institution, and will store that).